src/background_script/index.ts

```diff
@@ -68,4 +74,36 @@ browser.runtime.onMessage.addListener((request: any) => {
         return { error: err.message };
       });
   }
+
+  // Special cases
+  switch (request.type) {
```

Comment:
This switch needs to come before the `openPrompt` call above, otherwise `PROMPT_TYPE.INFO` opens a prompt. You can test this out by calling
```
webln.enable().then(() => webln.getInfo())
```
in your console.

Reply:
Oh wow, good catch. 

In my head, while refactoring this function, I intended to make the sender of `PROMPT_TYPE.INFO` messages set `prompt: false` on the message, since it shouldn't actually display a prompt. Clearly this slipped my mind.

I just looked into the usages of `PROMPT_TYPE.INFO` and realized that the sender of this message is in webln and would require a bit of refactoring to set `prompt: false`. So I'll make the update you suggested by putting the `switch` before the `openPrompt`.

Reply:
Great, thanks! There will be additional cases that could or would be prompts, but won't prompt the user (e.g. invoices that have a fixed amount can be auto-generated, payments that you have an allowance setup for can automatically be done) so there will still be things that are `prompt: true` that don't _actually_ prompt the user. I totally get where you were coming from though, a lot of this message passing stuff isn't very clear.

Reply:
Ok, I get it now. That makes sense then.

Reply:
I made the change requested above.